homeassistant/components/insteon_local.py

```diff
@@ -0,0 +1,49 @@
+"""
+Local Support for Insteon.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/insteon_local/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME, CONF_HOST)
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['insteonlocal']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'insteon_local'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Setup Insteon Hub component.
+
+    This will automatically import associated lights.
+    """
+    from insteonlocal.Hub import Hub
+
+    username = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = config[DOMAIN][CONF_HOST]
+
+    insteonhub = Hub(host, username, password)
+
+    if insteonhub is None:
+        _LOGGER.error("Could not connect to Insteon service")
+        return False
+
+    hass.data['insteon_local'] = insteonhub
+    discovery.load_platform(hass, 'light', DOMAIN, {}, config)
```

Comment:
I am actually confused because looking at the code, the platforms do not do anything with the discovery.

I would expect it to work like this: insteon_local will query for all available lights/switches and makes sure to fire off events for lights/switches to be setup. If a user specifies devices in their insteon_local component config, it will make sure to only fire for those devices.